pkg/kubectl/resource/mapper.go

```diff
@@ -62,10 +72,6 @@ func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
 		return nil, fmt.Errorf("unable to connect to a server to handle %q: %v", mapping.Resource, err)
 	}
 
-	// TODO: decoding the version object is convenient, but questionable. This is used by apply
-	// and rolling-update today, but both of those cases should probably be requesting the raw
-	// object and performing their own decoding.
-	obj, versioned := versions.Last(), versions.First()
```

Comment:
This didn't actually get removed, it just got moved up to line 61 in the `else` condition.